lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -67,7 +67,10 @@ def configure(conf)
       if @cache_ttl < 0
         @cache_ttl = :none
       end
-      @cache                                  = LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+      @cache = LruRedux::TTL::ThreadSafeCache.new(@cache_size, @cache_ttl)
+      if @include_namespace_id
+        @namespace_cache = LruRedux::TTL::ThreadSafeCache.new([25, @cache_size].min, @cache_ttl)
```

Comment:
Limiting the cache size to 25 send a bit low. We have multitenant use cases with 100s of namespaces & setting it this low would cause lots of api server requests & cache evictions. I'd probably just make 25 the default but allow users to configure without hard ceiling.